File: app/src/main/java/com/worldventures/dreamtrips/social/ui/membership/model/converter/InviteTemplateConverter.java
package com.worldventures.dreamtrips.social.ui.membership.model.converter;


import com.worldventures.core.converter.Converter;
import com.worldventures.dreamtrips.api.invitation.model.InvitationTemplate;
import com.worldventures.dreamtrips.social.ui.membership.model.InviteTemplate;

import io.techery.mappery.MapperyContext;

public class InviteTemplateConverter implements Converter<InvitationTemplate, InviteTemplate> {
   @Override
   public Class<InvitationTemplate> sourceClass() {
      return InvitationTemplate.class;
   }

   @Override
   public Class<InviteTemplate> targetClass() {
      return InviteTemplate.class;
   }

   @Override
   public InviteTemplate convert(MapperyContext mapperyContext, InvitationTemplate invitationTemplate) {
      InviteTemplate.Type type;
      switch (invitationTemplate.type()) {
         case SMS:
            type = InviteTemplate.Type.SMS;
            break;
         case EMAIL:
         case UNKNOWN:
         default:
            type = InviteTemplate.Type.EMAIL;
      }
      return new InviteTemplate(invitationTemplate.id(), invitationTemplate.title(), type,
            invitationTemplate.category(), invitationTemplate.coverImage().url(), invitationTemplate.video(),
            invitationTemplate.locale(), invitationTemplate.content());
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/domain/converter/InviteTemplateConverter.kt
package com.worldventures.dreamtrips.social.domain.converter


import com.worldventures.core.converter.Converter
import com.worldventures.dreamtrips.api.invitation.model.InvitationTemplate
import com.worldventures.dreamtrips.social.domain.entity.InviteTemplate

import io.techery.mappery.MapperyContext

class InviteTemplateConverter : Converter<InvitationTemplate, InviteTemplate> {

   override fun sourceClass() = InvitationTemplate::class.java

   override fun targetClass() = InviteTemplate::class.java

   override fun convert(context: MapperyContext, source: InvitationTemplate) =
         InviteTemplate(source.id(), source.title(), source.coverImage().url(), source.content(), source.category())
}
